pyomo/contrib/pynumero/algorithms/solvers/scipy_solvers.py

```diff
@@ -124,6 +126,141 @@ def solve(self, x0=None):
         return results
 
 
+class _ScalarDenseSquareNlpSolver(DenseSquareNlpSolver):
+    # A base class for solvers for scalar equations.
+    # Not intended to be instantiated directly. Instead,
+    # NewtonNlpSolver or SecantNewtonNlpSolver should be used.
+
+    def __init__(self, nlp, timer=None, options=None):
+        super().__init__(nlp, timer=timer, options=options)
+        if nlp.n_primals() != 1:
+            raise RuntimeError(
+                "Cannot use the scipy.optimize.newton solver on an NLP with"
+                " more than one variable and equality constraint. Got %s"
+                " primals. Please use RootNlpSolver or FsolveNlpSolver instead."
+            )
+
+
+class NewtonNlpSolver(_ScalarDenseSquareNlpSolver):
+    """A wrapper around the SciPy scalar Newton solver for NLP objects
+
+    """
+
+    OPTIONS = ConfigBlock(
+        description="Options for SciPy newton wrapper",
+    )
+    OPTIONS.declare("tol", ConfigValue(
+        default=1e-8,
+        domain=float,
+        description="Convergence tolerance",
+    ))
+    OPTIONS.declare("secant", ConfigValue(
+        default=False,
+        domain=bool,
+        description="Whether to use SciPy's secant method",
+    ))
+    OPTIONS.declare("full_output", ConfigValue(
+        default=True,
+        domain=bool,
+        description="Whether underlying solver should return its full output",
+    ))
+    OPTIONS.declare("maxiter", ConfigValue(
+        default=50,
+        domain=int,
+        description="Maximum number of function evaluations per solve",
+    ))
+
+    def solve(self, x0=None):
+        self._timer.start("solve")
+        if x0 is None:
+            x0 = self._nlp.get_primals()
+
+        if self.options.secant:
+            fprime = None
+        else:
+            fprime = lambda x: self.evaluate_jacobian(np.array([x]))[0, 0]
+        results = sp.optimize.newton(
+            lambda x: self.evaluate_function(np.array([x]))[0],
+            x0[0],
+            fprime=fprime,
+            tol=self.options.tol,
+            full_output=self.options.full_output,
+            maxiter=self.options.maxiter,
+        )
+        self._timer.stop("solve")
+        return results
+
+
+class SecantNewtonNlpSolver(_ScalarDenseSquareNlpSolver):
```

Comment:
Why does this not derive from the `NewtonNLPSolver`?

Reply:
These solvers support different options, and I wasn't aware of  `declare_from` with `skip=...`. Otherwise there is not a huge amount of repeated code. Don't think I have an excuse for PyomoSecantNewtonSolver below.